presto-hive/src/main/java/io/prestosql/plugin/hive/ViewReaderUtil.java

```diff
@@ -0,0 +1,229 @@
+                    log.debug("Adding column %s, %s, %s, %s", field.getName(),
+                            field.getType().getFullTypeString(),
+                            field.getType().getSqlTypeName(),
+                            field);
+
+                    Type type = typeManager.fromSqlType(getTypeString(field.getType()));
+                    ConnectorViewDefinition.ViewColumn column = new ConnectorViewDefinition.ViewColumn(field.getName(), type.getTypeId());
+                    columns.add(column);
+                }
+                return new ConnectorViewDefinition(prestoSql,
+                        Optional.of(catalogName.toString()),
+                        Optional.of(table.getDatabaseName()),
+                        columns,
+                        Optional.ofNullable(table.getParameters().get(TABLE_COMMENT)),
+                        Optional.empty(),
+                        true);
+            }
+            catch (RuntimeException e) {
+                throw new PrestoException(HIVE_UNKNOWN_ERROR,
+                        format("Error decoding view definition for %s.%s: %s. Please report this to ask_dali@linkedin.com",
+                                table.getDatabaseName(),
+                                table.getTableName(),
+                                e.getMessage()),
+                        e);
+            }
+        }
+
+        // Calcite does not provide correct type strings for non-primitive types.
+        // We add custom code here to make it work. Goal is for calcite/coral to handle this
+        private String getTypeString(RelDataType type)
+        {
+            switch (type.getSqlTypeName()) {
+                case ROW: {
+                    Preconditions.checkState(type.isStruct());
+                    StringBuilder sb = new StringBuilder("row(");
+                    List<RelDataTypeField> fieldList = type.getFieldList();
+                    for (int i = 0; i < fieldList.size(); i++) {
+                        if (i != 0) {
+                            // There should not be space after comma. Presto deserializer doesn't like it
+                            sb.append(",");
+                        }
+                        RelDataTypeField field = fieldList.get(i);
+                        sb.append(field.getName().toLowerCase(Locale.ENGLISH)).append(" ").append(getTypeString(field.getType()));
+                    }
+                    sb.append(")");
+                    return sb.toString();
+                }
+                case CHAR:
+                    return "varchar";
+                case FLOAT:
+                    return "real";
+                case BINARY:
+                case VARBINARY:
+                    return "varbinary";
+                case MAP: {
+                    RelDataType keyType = type.getKeyType();
+                    RelDataType valueType = type.getValueType();
+                    StringBuilder sb = new StringBuilder("map(");
```

Comment:
This can chain (no need for a variable). Though I think it's easier to read with string formatting
```java
return format("map(%s,%s)", getTypeString(keyType), getTypeString(valueType));